src/App.jsx

```diff
@@ -25,22 +26,24 @@ const App = () => {
       <MuiThemeProvider theme={theme}>
         <ScThemeProvider theme={theme}>
           <GlobalStyle />
-          <Layout>
-            <Routes>
-              <Route path='/' element={<HomePage />} />
-              <Route path='/post/new' element={<CreatePostPage />} />
-              <Route path='/post/:postId' element={<PostDetailPage />} />
-              <Route path='/hotel/:hotelId' element={<HotelDetailPage />} />
-              <Route path='/search' element={<SearchPage />} />
-              <Route path='/login' element={<LoginPage />} />
-              <Route path='/signup' element={<SignupPage />} />
-              <Route path='/user/profile' element={<ProfilePage />} />
-              <Route path='/user/:userId' element={<UserPage />} />
-              <Route path='/about' element={<AboutPage />} />
-              <Route path='/develop' element={<DevelopPage />} />
-              <Route path='/hotel/new' element={<CreateHotelPage />} />
-            </Routes>
-          </Layout>
+          <RecoilRoot>
+            <Layout>
+              <Routes>
+                <Route path='/' element={<HomePage />} />
+                <Route path='/post/new' element={<CreatePostPage />} />
+                <Route path='/post/:postId' element={<PostDetailPage />} />
+                <Route path='/hotel/:hotelId' element={<HotelDetailPage />} />
+                <Route path='/search' element={<SearchPage />} />
+                <Route path='/login' element={<LoginPage />} />
+                <Route path='/signup' element={<SignupPage />} />
+                <Route path='/user/profile' element={<ProfilePage />} />
+                <Route path='/user/:userId' element={<UserPage />} />
+                <Route path='/about' element={<AboutPage />} />
+                <Route path='/develop' element={<DevelopPage />} />
+                <Route path='/hotel/new' element={<CreateHotelPage />} />
+              </Routes>
+            </Layout>
+          </RecoilRoot>
```

Comment:
```suggestion
            <Layout>
              <Routes>
                <Route path='/' element={<HomePage />} />
                <Route path='/post/new' element={<CreatePostPage />} />
                <Route path='/post/:postId' element={<PostDetailPage />} />
                <Route path='/hotel/:hotelId' element={<HotelDetailPage />} />
                <Route path='/search' element={<SearchPage />} />
                <Route path='/login' element={<LoginPage />} />
                <Route path='/signup' element={<SignupPage />} />
                <Route path='/user/profile' element={<ProfilePage />} />
                <Route path='/user/:userId' element={<UserPage />} />
                <Route path='/about' element={<AboutPage />} />
                <Route path='/develop' element={<DevelopPage />} />
                <Route path='/hotel/new' element={<CreateHotelPage />} />
              </Routes>
            </Layout>
```
RecoilRootはindex.jsに書いてるのでApp.jsxでは不要です。
要らなければ削除しておいてください。